pkg/steps/release/promote.go

```diff
@@ -94,25 +94,32 @@ func (s *promotionStep) run(ctx context.Context) error {
 		return fmt.Errorf("could not resolve pipeline imagestream: %w", err)
 	}
 
-	imageMirrorTarget, namespaces := getImageMirrorTarget(tags, pipeline, registryDomain(s.configuration.PromotionConfiguration))
+	date := time.Now().Format("20060102")
+	imageMirrorTarget, namespaces := getImageMirrorTarget(tags, pipeline, registryDomain(s.configuration.PromotionConfiguration), date, s.mirrorFunc)
+	if s.name == api.PromotionQuayStepName {
```

Comment:
The difference here is just the target registry, it should be another parameter of the step.